src/ray/core_worker/task_event_buffer.h

```diff
@@ -33,6 +33,45 @@ namespace core {
 
 namespace worker {
 
+/// A  wrapper class that will be converted to rpc::TaskEvents
+///
+/// This will be created by CoreWorker and stored in TaskEventBuffer, and
+/// when it is being flushed periodically to GCS, it will be converted to
+/// rpc::TaskEvents.
+/// This is an optimization so that converting to protobuf (which is costly)
+/// will not happen in the critical path of task execution/submission.
+class TaskEvent {
+ public:
```

Comment:
Can we make these data fields const + populated in a constructor? This is a lot of mutable state..